### PR TITLE
Moved IAM Policy from here string to IAM Data Block

### PR DESCRIPTION
**Pull request by wesleykirklandsg**
Migrate from IAM policy here string to IAM Data Block

### CODE REVIEW

**Comment by wesleykirklandsg** [comment]
@rchildress87 

Original Policy Evaluated in AWS
```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "ListUsersAccessWithoutMFA",
            "Action": [
                "iam:ListVirtualMFADevices"
            ],
            "Effect": "Allow",
            "Resource": "*"
        },
        {
            "Sid": "SelfServiceAccessWithoutMFA",
            "Action": [
                "iam:List*",
                "iam:GetUser",
                "iam:GetAccountPasswordPolicy",
                "iam:ChangePassword",
                "iam:CreateVirtualMFADevice",
                "iam:EnableMFADevice"
            ],
            "Effect": "Allow",
            "Resource": [
                "arn:aws:iam::*:user/${aws:username}",
                "arn:aws:iam::*:mfa/${aws:username}"
            ]
        },
        {
            "Sid": "SelfServiceAccessWithMFA",
            "Effect": "Allow",
            "Action": [
                "iam:Get*",
                "iam:DeleteSSHPublicKey",
                "iam:GetSSHPublicKey",
                "iam:ListSSHPublicKeys",
                "iam:UpdateSSHPublicKey",
                "iam:UploadSSHPublicKey",
                "iam:CreateAccessKey",
                "iam:DeleteAccessKey",
                "iam:UpdateAccessKey",
                "iam:DeleteVirtualMFADevice",
                "iam:DeactivateMFADevice",
                "iam:ResyncMFADevice",
                "iam:UploadSigningCertificate",
                "iam:UpdateSigningCertificate",
                "iam:DeleteSigningCertificate",
                "iam:GenerateServiceLastAccessedDetails"
            ],
            "Resource": [
                "arn:aws:iam::*:user/${aws:username}",
                "arn:aws:iam::*:mfa/${aws:username}"
            ],
            "Condition": {
                "BoolIfExists": {
                    "aws:MultiFactorAuthPresent": true
                }
            }
        }
    ]
}
```

TF Evaluated Policy in AWS
```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "ListUsersAccessWithoutMFA",
            "Effect": "Allow",
            "Action": "iam:ListVirtualMFADevices",
            "Resource": "*"
        },
        {
            "Sid": "SelfServiceAccessWithoutMFA",
            "Effect": "Allow",
            "Action": [
                "iam:List*",
                "iam:GetUser",
                "iam:GetAccountPasswordPolicy",
                "iam:EnableMFADevice",
                "iam:CreateVirtualMFADevice",
                "iam:ChangePassword"
            ],
            "Resource": [
                "arn:aws:iam::*:user/${aws:username}",
                "arn:aws:iam::*:mfa/${aws:username}"
            ]
        },
        {
            "Sid": "SelfServiceAccessWithMFA",
            "Effect": "Allow",
            "Action": [
                "iam:UploadSigningCertificate",
                "iam:UploadSSHPublicKey",
                "iam:UpdateSigningCertificate",
                "iam:UpdateSSHPublicKey",
                "iam:UpdateAccessKey",
                "iam:ResyncMFADevice",
                "iam:ListSSHPublicKeys",
                "iam:GetSSHPublicKey",
                "iam:Get*",
                "iam:GenerateServiceLastAccessedDetails",
                "iam:DeleteVirtualMFADevice",
                "iam:DeleteSigningCertificate",
                "iam:DeleteSSHPublicKey",
                "iam:DeleteAccessKey",
                "iam:DeactivateMFADevice",
                "iam:CreateAccessKey"
            ],
            "Resource": [
                "arn:aws:iam::*:user/${aws:username}",
                "arn:aws:iam::*:mfa/${aws:username}"
            ],
            "Condition": {
                "BoolIfExists": {
                    "aws:MultiFactorAuthPresent": "true"
                }
            }
        }
    ]
}
```

File: iamdata.tf [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    effect = "Allow"
+
+    resources = [
+      "arn:aws:iam::*:user/$${aws:username}",
+      "arn:aws:iam::*:mfa/$${aws:username}"
+    ]
+
+    condition {
+      test     = "BoolIfExists"
```
Comment (rchildress87) [inline iamdata.tf:66]: I think this maybe should be `bool`
https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_condition-keys.html#condition-keys-multifactorauthpresent

**Review by rchildress87 — CHANGES_REQUESTED** [review]

File: iamdata.tf (hunk 1 above)
Comment (wesleykirklandsg) [reply iamdata.tf:66]: @rchildress87 That document says `BoolIfExists` is correct, the policy we're using is the AWS published one that everyone uses.

We recommend that you use the [BoolIfExists](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_condition_operators.html#Conditions_IfExists) operator to check whether a request is authenticated using MFA.

Comment (rchildress87) [reply iamdata.tf:66]: I think one of the `bool` example under Alternative Combinations should be considered as better option but, according to above documentation, this does emulate the policy that is created when you use the IAM visual editor and choose *MFA Required*. It also matches the policy that this data source is replacing. I'll approve.

**Review by rchildress87 — APPROVED** [review]

**Pull request merged**